### ISSUE TITLE
KryoException: Encountered unregistered class ID: 71

### ISSUE THREAD

**Issue by pihme** [camunda-cloud/zeebe#5506]
**Describe the bug**
Observed as error in production logs.

**Log/Stacktrace**
https://console.cloud.google.com/errors/CN6-vs-5n-TG1QE?service=zeebe&time=P7D&project=camunda-cloud-240911

**Environment:**
Camunda Cloud / Prod / 0.24.2

**Comment by saig0** [comment]
Happened again on `0.24.4`: (https://console.cloud.google.com/errors/CKKRstaF1ZWPew?service=zeebe&time=P7D&project=camunda-cloud-240911)

```
com.esotericsoftware.kryo.KryoException: Encountered unregistered class ID: 75
at com.esotericsoftware.kryo.util.DefaultClassResolver.readClass (DefaultClassResolver.java:137)
at com.esotericsoftware.kryo.Kryo.readClass (Kryo.java:693)
at com.esotericsoftware.kryo.Kryo.readClassAndObject (Kryo.java:804)
at io.atomix.utils.serializer.NamespaceImpl.deserialize (NamespaceImpl.java:214)
at io.atomix.utils.serializer.FallbackNamespace.deserialize (FallbackNamespace.java:113)
at io.atomix.storage.journal.FileChannelJournalSegmentReader.readNextEntry (FileChannelJournalSegmentReader.java:215)
at io.atomix.storage.journal.FileChannelJournalSegmentReader.readNext (FileChannelJournalSegmentReader.java:197)
at io.atomix.storage.journal.FileChannelJournalSegmentReader.next (FileChannelJournalSegmentReader.java:117)
at io.atomix.storage.journal.MappableJournalSegmentReader.next (MappableJournalSegmentReader.java:113)
at io.atomix.storage.journal.SegmentedJournalReader.next (SegmentedJournalReader.java:119)
at io.atomix.storage.journal.DelegatingJournalReader.next (DelegatingJournalReader.java:66)
at io.zeebe.logstreams.storage.atomix.AtomixLogStorageReader.findEntry (AtomixLogStorageReader.java:156)
at io.zeebe.logstreams.storage.atomix.AtomixLogStorageReader.read (AtomixLogStorageReader.java:55)
at io.zeebe.logstreams.impl.log.LogStreamReaderImpl.lambda$readBlockIntoBuffer$1 (LogStreamReaderImpl.java:193)
at io.zeebe.logstreams.impl.log.LogStreamReaderImpl.executeReadMethod (LogStreamReaderImpl.java:197)
at io.zeebe.logstreams.impl.log.LogStreamReaderImpl.readBlockIntoBuffer (LogStreamReaderImpl.java:192)
at io.zeebe.logstreams.impl.log.LogStreamReaderImpl.readNextAddress (LogStreamReaderImpl.java:225)
at io.zeebe.logstreams.impl.log.LogStreamReaderImpl.readNextEvent (LogStreamReaderImpl.java:243)
at io.zeebe.logstreams.impl.log.LogStreamReaderImpl.next (LogStreamReaderImpl.java:176)
at io.zeebe.logstreams.impl.log.LogStreamReaderImpl.next (LogStreamReaderImpl.java:20)
at io.zeebe.engine.processor.ReProcessingStateMachine.scanLog (ReProcessingStateMachine.java:162)
at io.zeebe.engine.processor.ReProcessingStateMachine.startRecover (ReProcessingStateMachine.java:139)
at io.zeebe.engine.processor.StreamProcessor.onActorStarted (StreamProcessor.java:113)
at io.zeebe.util.sched.ActorJob.invoke (ActorJob.java:73)
at io.zeebe.util.sched.ActorJob.execute (ActorJob.java:39)
at io.zeebe.util.sched.ActorTask.execute (ActorTask.java:118)
at io.zeebe.util.sched.ActorThread.executeCurrentTask (ActorThread.java:107)
at io.zeebe.util.sched.ActorThread.doWork (ActorThread.java:91)
at io.zeebe.util.sched.ActorThread.run (ActorThread.java:204)
```

From the logs (https://console.cloud.google.com/logs/query;query=%0AlogName:%22stdout%22%0Aresource.type%3D%22k8s_container%22%0Aresource.labels.cluster_name%3D%22ultratest%22%0Aresource.labels.namespace_name%3D%228235f864-3a5b-4cbe-a9f6-a335d478fb26-zeebe%22%0Aresource.labels.container_name%3D%22zeebe%22%0A-jsonPayload.logger%3D%22io.zeebe.broker.workflow.repository%22%0Aresource.labels.pod_name%3D%22zeebe-2%22;timeRange=2020-10-18T07:55:11.093Z%2F2020-10-18T08:55:11.093Z;pinnedLogId=2020-10-18T08:24:41.093285Z%2Faynbv44o0fz1dr6n4?project=camunda-cloud-240911):
```
Debug
2020-10-18 10:23:39.339 CEST
"Closing Zeebe-Partition-8: stream processor"
Warning
2020-10-18 10:23:40.722 CEST
"RaftServer{raft-partition-partition-2}{role=FOLLOWER} - Poll request to 1 failed: java.util.concurrent.TimeoutException: Request type raft-partition-partition-2-poll timed out in 5000 milliseconds"
Debug
2020-10-18 10:23:46.284 CEST
"Received metadata change from Broker 3, partitions {2=FOLLOWER, 3=FOLLOWER, 4=FOLLOWER, 8=FOLLOWER} and terms {}."
Debug
2020-10-18 10:23:46.289 CEST
"Received metadata change for 3, partitions {2=FOLLOWER, 3=FOLLOWER, 4=FOLLOWER, 8=FOLLOWER} terms {}"
Warning
2020-10-18 10:23:48.429 CEST
"RaftServer{raft-partition-partition-2}{role=FOLLOWER} - Poll request to 1 failed: java.util.concurrent.TimeoutException: Request type raft-partition-partition-2-poll timed out in 5000 milliseconds"
Warning
2020-10-18 10:23:50.151 CEST
"RaftServer{raft-partition-partition-2}{role=FOLLOWER} - Poll request to 3 failed: io.netty.channel.ConnectTimeoutException: connection timed out: zeebe-3.zeebe-broker-service.8235f864-3a5b-4cbe-a9f6-a335d478fb26-zeebe.svc.cluster.local/10.28.10.12:26502"
Warning
2020-10-18 10:23:50.165 CEST
"RaftServer{raft-partition-partition-2}{role=FOLLOWER} - Poll request to 3 failed: io.netty.channel.ConnectTimeoutException: connection timed out: zeebe-3.zeebe-broker-service.8235f864-3a5b-4cbe-a9f6-a335d478fb26-zeebe.svc.cluster.local/10.28.10.12:26502"
Debug
2020-10-18 10:23:56.046 CEST
"Received metadata change for 1, partitions {1=FOLLOWER, 2=LEADER, 6=FOLLOWER, 7=FOLLOWER, 8=LEADER} terms {2=488497, 8=509402}"
Debug
2020-10-18 10:23:56.047 CEST
"Received metadata change from Broker 1, partitions {1=FOLLOWER, 2=LEADER, 6=FOLLOWER, 7=FOLLOWER, 8=LEADER} and terms {2=488497, 8=509402}."
Debug
2020-10-18 10:24:02.093 CEST
"Received metadata change for 0, partitions {1=FOLLOWER, 5=FOLLOWER, 6=LEADER, 7=LEADER} terms {6=18688, 7=7265}"
Debug
2020-10-18 10:24:02.097 CEST
"Received metadata change from Broker 0, partitions {1=FOLLOWER, 5=FOLLOWER, 6=LEADER, 7=LEADER} and terms {6=18688, 7=7265}."
Debug
2020-10-18 10:24:15.312 CEST
"Partition role transitioning from CANDIDATE to FOLLOWER"
Debug
2020-10-18 10:24:20.368 CEST
"Received metadata change for 0, partitions {1=LEADER, 5=FOLLOWER, 6=LEADER, 7=LEADER} terms {1=3515, 6=18688, 7=7265}"
Debug
2020-10-18 10:24:20.370 CEST
"Received metadata change from Broker 0, partitions {1=LEADER, 5=FOLLOWER, 6=LEADER, 7=LEADER} and terms {1=3515, 6=18688, 7=7265}."
Error
2020-10-18 10:24:41.093 CEST
"Actor Broker-2-StreamProcessor-8 failed in phase STARTED."
```

It seems to be a race condition between starting and closing.

**Comment by npepinpe** [comment]
I'm closing this for now - while it's not fixed for < 1.0 versions, no users or customers have ever reported issues related to this, and we've dropped Kryo completely from our storage by now. I would not spend time on this until users are also reporting it, especially considering we will drop support for pre 1.0 in July/August - I doubt we will fix this until then unless it affects more users on older versions.
